### PR TITLE
Serialise VSI to Mantid project files

### LINKED ISSUE #17368: Serialise VSI
This is a sub issue of #15602 .

Serialise settings on the VSI to a Mantid project file.

### PR DESCRIPTION
**Pull request by samueljackson92**
**This should only be merged after #16961 (Merged 19 Sept 2016)**

When saving a Mantid project, VSI will now dump it's state to a file called `VSI.xml`.  This file contains an XML representation of the camera position, colours, surface type, views, sources etc. and is reloaded when the project is opened.

**To test:**
You need to test all of the different views in VSI, with both single and multiple sources for peaks workspaces, MDEvent and MDHisto workspaces. Reloading the project should leave VSI looking just the same as when it was saved.

Release notes will be handled in #17109 .

Fixes #17368 .

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Vates/VatesSimpleGui/ViewWidgets/src/StandardView.cpp [hunk 1]
```diff
@@ -287,6 +291,23 @@ void StandardView::updateView() { this->m_cameraReset = true; }
...
+  QHBoxLayout *hbox = new QHBoxLayout(this->m_ui.renderFrame);
+  hbox->setMargin(0);
+  hbox->addWidget(m_view->widget());
+
+  QObject::connect(this->m_view.data(), SIGNAL(endRender()), this,
+                   SLOT(onRenderDone()));
+}
+
+ModeControlWidget::Views StandardView::getViewType() {
+  return ModeControlWidget::Views::STANDARD;
```
Comment (AntonPiccardoSelg) [inline Vates/VatesSimpleGui/ViewWidgets/src/StandardView.cpp]: This is a good approach :+1: . Should save quite a bit of dynamic casting that was going on previously.

Comment (samueljackson92) [inline Vates/VatesSimpleGui/ViewWidgets/src/StandardView.cpp]: 👍

**Comment by AntonPiccardoSelg** [comment]
@samueljackson92  Tried to save a project with a sample workspace on Windows. Reopening it prodces a PV error. It cannot find the VSI.xml file. Looking into the project file, there does not seem to exist one.

**Comment by AndreiSavici** [comment]
I've added the `In Progress` label until the issues found by @AntonPiccardoSelg are fixed

**Comment by AntonPiccardoSelg** [comment]
All major issues have been resolved. The serialization of the VSI state is a huge step foward. There are very few minor things which we might have to address at some point, but they are very minor and should  not hold back the publication of this work.

Again well done and happy to :shipit:

**Review by martyngigg — APPROVED** [review]

**Pull request merged**